internal/globalconfig/globalconfig.go

```diff
@@ -62,3 +70,41 @@ func RuntimeID() string {
 	defer cfg.mu.RUnlock()
 	return cfg.runtimeID
 }
+
+// HTTPClientCodes returns the http client codes identified as errors.
+func HTTPClientCodes() *bitset.BitSet {
+	cfg.mu.RLock()
+	defer cfg.mu.RUnlock()
+	return cfg.httpClientCodes
+}
+
+// SetHTTPClientCodes sets the http client codes identified as errors.
+func SetHTTPClientCodes(codes *bitset.BitSet) {
+	cfg.mu.Lock()
+	defer cfg.mu.Unlock()
+	cfg.httpClientCodes = codes
+}
+
+// HTTPServerCodes returns the http server codes identified as errors.
+func HTTPServerCodes() *bitset.BitSet {
```

Comment:
Note that we don't want this API, since it's only purpose is for testing. However, you could obtain what you want (from a testing standpoint) if you moved all the parsing logic here, by basically changing `SetHTTP*Codes` to take a string (same thing that `parseCodes` does now) and do the parsing here. Then, as a result, you could have your tests here, and you'd be able to access the private parts too (or make the getter private).

We should not, in general, ever create exported API (even if internal) just for the sake of facilitating tests, unless it's a specific testing package.